src/models/test/BrowserStyleSheet.test.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+
+import React from 'react'
+import { mount } from 'enzyme'
+
+import styled from '../../index'
+import BrowserStyleSheet from '../BrowserStyleSheet'
+import StyleSheetManager from '../StyleSheetManager'
+import { expectCSSMatches, resetStyled } from '../../test/utils'
+
+describe('BrowserStyleSheet', () => {
+  it('should render styles in correct order when styled(StyledComponent) and StyleSheetManager is used', () => {
+    const Red = styled.div`
+      color: red;
+    `
+
+    const RedChangedToBlue = styled(Red)`
+      color: blue;
+    `
+
+    const sheet = BrowserStyleSheet.create()
+
+    const App = () =>
+      <StyleSheetManager sheet={sheet}>
```

Comment:
So: I'm manually passing `BrowserStyleSheet` to `StyleSheetManager`. This is probably valid way to catch a bug in #1102, since that behaviour is identical to what that pull request essentially does, but I'm unsure whether using `StyleSheetManager` in this manner says anything about #1164.